File: backend/app/schemas/audit_logs.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Any
from datetime import datetime

class AuditLogItem(BaseModel):
  id: int
  action: str
  entity_type: Optional[str] = None
  entity_id: Optional[Any] = None
  details: Optional[Any] = None
  ip_address: Optional[str] = None
  created_at: datetime

  model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/auditLogs.ts
import apiClient from './client';
import { PaginatedResponse } from '../types/common';

export interface AuditLog {
  id: number;
  action: string;
  entity_type: string;
  entity_id: number | null;
  details: any;
  ip_address: string | null;
  created_at: string;
}

export const getAuditLogs = async (params?: {
  page?: number;
  page_size?: number;
  action?: string;
  entity_type?: string;
  date_from?: string;
  date_to?: string;
}) => {
  const { data } = await apiClient.get<PaginatedResponse<AuditLog>>('/audit-logs', { params });
  return data;
};
